### ISSUE TITLE
Add support for Layers in FBX

### ISSUE THREAD

**Issue by Nodrev** [assimp/assimp#3919]
3DSMax have the notion of "Layers" to organize nodes by group (not in a hierarchical way), and Blender have the same principle named "Collections".

I need to generate FBX files with nodes associated to layers. It's pretty straightforward in FBX side (extracts are from `test/models/FBX/box_orphant_embedded_texture.fbx`):
* Declare a new object type in the header:
```
	ObjectType: "CollectionExclusive" {
		Count: 1
		PropertyTemplate: "FbxDisplayLayer" {
			Properties70:  {
				P: "Color", "ColorRGB", "Color", "",0.8,0.8,0.8
				P: "Show", "bool", "", "",1
				P: "Freeze", "bool", "", "",0
				P: "LODBox", "bool", "", "",0
			}
		}
	}
```
* Add a "CollectionExclusive / DisplayLayer" object per layer
```
	CollectionExclusive: 2959111896352, "DisplayLayer::Box", "DisplayLayer" {
		Properties70:  {
			P: "Color", "ColorRGB", "Color", "",0.607999980449677,0,0.157000005245209
		}
	}
```
* Associate nodes to layers, and link sub-layers togethers:
```
	;Model::Box, DisplayLayer::Box
	C: "OO",2957618625584,2959111896352
```

So, pretty easy, neah? The problem I have is how to handle this additional information on Assimp side. I see two solutions:
* The simpler/faster is to add a custom metadata to aiNode (need a common metadata id for this?)
* Add a `aiLayer` object to API, and add a `aiLayer* rootLayer` object to `aiScene`, it's a lot more work and a potential API break :/

As you see, neither of the 2 proposal are perfect, what do you think?

**Comment by kimkulling** [comment]
Seems to be a new layer of access. At this moment we are supporting a basic API to work with the render data with all the related data to transformation and animations. 

And this feature request seems to be related to the object access layer.

**Comment by RichardTea** [comment]
As I understand this, every scene node would need to have both a parent (null for the root) and a layer (can be null for all)

However there would _also_ be a need for an extra "Layer" object type to contain data about the layer - its name, colour, and other application-specific information.
Otherwise there's nowhere to store that global layer data.

So it would definitely need to be a binary breaking change as the Layer object would need to be introduced.

That said, the current HEAD is no longer binary compatible with 5.0.1 anyway, so perhaps now is a good time to do this?

**Comment by kimkulling** [comment]
Or just an enhancement for the current API. So no existing API would get a break. Just a new layer introduced by some kind of aiObject with its layers.

**Comment by kimkulling** [comment]
Such a layer would not break the backward compatibility.

**Comment by Nodrev** [comment]
So as I was thinking, the API choice is not obvious :).
Let me know when your mind will be clear about the solution you prefer!

**Comment by kimkulling** [comment]
Indeed, this API solution seems not to be ovious. But a lot of asset formats are storing object grouping information in the data. And this would not break the backward compatibility. And the aiScene-Data will not get blown up.

**Comment by kimkulling** [comment]
Sometimes you have to think big, you know? :-):-):-)

**Comment by tellypresence** [comment]
Please change issue title to "Add support for Layers in FBX" (remove "export" to avoid confusion with assimp export feature)

**Comment by tellypresence** [comment]
@Nodrev Is this still a problem?

**Comment by tellypresence** [comment]
Closing due to no response to followup

**Comment by kimkulling** [comment]
This is still a good idea.
